scripts/legislation/africa/rwanda_french.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+function remove_all_text_before_first_chapter_header {
+  sed -n '/CHAPITRE PREMIER - DES DISPOSITIONS GENERALES /,$p'
+}
+
+function amend_errors_in_headers {
+  sed -E 's/^(Section [0-9]+ - [A-Za-z,’ ]+)[^a-z]+Article ([0-9]+)(er)?/\1\n\n(\2)/' | \
+    sed -E 's/Article premier:/\n\n(1)/' | \
+    sed -E 's/^(CHAPITRE )I -l:/\1II -/' | \
+    sed -E 's/^(TITRE )I -l:/\1II -/' | \
+    sed -E "s/(CHAPITRE PREMIER - DE L'ENVIRONNEMENT NATUREL )/\1\n\n/" | \
+    sed -E 's/^(Section première: Le sol.*) Article 11:/\1\n\n(11)/' | \
+    sed -E 's/^(Section 3 - De la diversité.*) Article 20:/\1\n\n(20)/' | \
+    sed -E "s/^(Section 4 - De l'atmosphère.*) Article 25:/\1\n\n(25)/" | \
+    sed -E "s/^(Section 2 - Obligations.*) Article 60:/\1\n\n(60)/" | \
+    sed -E 's/^(CHAPITRE )I -ll:/\1III -/' 
+}
+
+function reformat_numbering_within_articles {
+  sed -E 's/([0-9])(°)/[\1]/g' 
```

Comment:
In general (although there are some exceptions), you only need to capture matching text if you plan to use it in your replacement text. As such, capturing the ° in this case is unnecessary.

Reply:
The text I am trying to replace is something like this `quiconque: 1° fait` within an article. If I remove the '(°)' in the function the text becomes quite problematic. From my understanding this function matches a number (capture group 1) which is followed by '°' (capture group 2) and replaces it with just the number (capture group 1) in square brackets.

Reply:
Agreed. I'm just saying that you don't need to capture the degree symbol because you don't use it in your replacement text. Instead, you should be able to write:

```
sed -E 's/([0-9])°/[\1]/g'
```

Reply:
Great, thanks for the tip. I did not know you could use the capture group like that.